internal/backend/local/test.go

```diff
@@ -74,12 +71,8 @@ type TestSuiteRunner struct {
 	// Verbose tells the runner to print out plan files during each test run.
 	Verbose bool
 
-	// configProviders is a cache of config keys mapped to all the providers
-	// referenced by the given config.
-	//
-	// The config keys are globally unique across an entire test suite, so we
-	// store this at the suite runner level to get maximum efficiency.
-	configProviders map[string]map[string]bool
+	Concurrency int
+	semaphore   terraform.Semaphore
```

Comment:
We could probably just let whoever creates the runner also creates the semaphore directly since that's all the Concurrency is doing?

Reply:
I think it's fine for the caller to just give us the concurrency value, and whether we use a semaphore or some other mechanism, is entirely an implementation detail.